src/complexity_measures/statistical_complexity.jl

```diff
@@ -0,0 +1,197 @@
+
+[^Rosso2007]: Rosso, O. A. et al. (2007). Distinguishing Noise from Chaos. 
+    Physical Review Letters 99, no. 15: 154102. https://doi.org/10.1103/PhysRevLett.99.154102.
+[^Rosso2013]: Rosso, O. A. (2013) Generalized Statistical Complexity: A New Tool for Dynamical Systems.
+
+"""
+Base.@kwdef struct StatisticalComplexity{E, D, H} <: ComplexityEstimator
+    dist::D = JSDivergence()
+    est::E = SymbolicPermutation()
+    entr::H = Renyi()
+
+    function StatisticalComplexity(dist::D, est::E, entr::H) where {E, D, H}
+        new{E, D, H}(dist, est, entr)
+    end
+    function StatisticalComplexity(x::AbstractVector;
+            dist::SemiMetric, est::ProbabilitiesEstimator,
+            entr::EntropyDefinition)
+        StatisticalComplexity(dist, est, entr)
+    end
+end
+
+function complexity(c::StatisticalComplexity, x::AbstractVector{T}) where T
+    (; dist, est, entr) = c
+
+    p = probabilities(est, x)
+
+    H_q = entropy_normalized(entr, est, x)
+
+    L = total_outcomes(est, x)
+
+    # calculate distance between calculated distribution and uniform one
+    D_q = evaluate(dist, p.p, fill(1.0/L, size(p)))
+
+    # generate distribution with just one filled bin
+    deterministic = zeros(size(p))
+    deterministic[1] = 1
+
+    D_max = evaluate(dist, deterministic, fill(1.0/L, size(p)))
+    C_q = D_q / D_max * H_q
+
+    return C_q
+end
+
+function complexity(c::StatisticalComplexity, p::Probabilities)
+    (; dist, est, entr) = c
+    if !(est isa SymbolicPermutation || est isa SymbolicWeightedPermutation || est isa SymbolicAmplitudeAwarePermutation)
+        throw(
+            ArgumentError(
+                "To estimate the statistical complexity directly from a probabiliy distribution, "*
+                "the probabilities estimator must be one of"*
+                "`SymbolicPermutation`, `SymbolicAmplitudeAwarePermutation` or `SymbolicWeightedPermutation`"
+            )
+        )
+    end
+    L = total_outcomes(est, randn(100))
+    norm = log(entr.base, L)
+    H_q = entropy(entr, p) / norm
+
+    # calculate distance between calculated distribution and uniform one
+    D_q = evaluate(dist, p.p, fill(1.0/L, size(p)))
+
+    # generate distribution with just one filled bin
+    deterministic = zeros(size(p))
+    deterministic[1] = 1
+
+    D_max = evaluate(dist, deterministic, fill(1.0/L, size(p)))
+    C_q = D_q / D_max * H_q
+
+    return C_q
+end
+
+function fill_probs_k!(p, prob_params, L, i, k)
+    p.p[1] = prob_params[k] # why set first element here if overwriting it in the loop below?
+    # if we know that p has sufficient length relative to L and i,
+    # @inbounds can save some computation time by skipping bounds checking.
+    @inbounds for j = 1:(L - i) 
+        p.p[j] = (1 - prob_params[k]) / (L - i)
+    end
+end
+
+linearpermissiverange(start; stop, length) = length==1 ? (start:start) : range(start, stop=stop, length=length)
+
+"""
+    maximum_complexity_entropy(c::StatisticalComplexity; num=1)
```

Comment:
It's not clear to me from this docstring what `num` is. Is is a user-determined constant? In any case, it should have a more informative name?